XenAdmin/TabPages/ManageUpdatesPage.cs

```diff
@@ -669,7 +670,45 @@ private void StoreSelectedUpdates()
 
         private void UpdateButtonEnablement()
         {
-            toolStripButtonExportAll.Enabled = toolStripSplitButtonDismiss.Enabled = Updates.UpdateAlertsCount > 0;            
+            toolStripButtonExportAll.Enabled = toolStripSplitButtonDismiss.Enabled = Updates.UpdateAlertsCount > 0;
+            toolStripButtonUpdate.ToolTipText = "";
+            var connectionList = ConnectionsManager.XenConnectionsCopy;
+
+            if (!connectionList.Any(xenConnection => xenConnection.IsConnected))
+            {
+                toolStripButtonUpdate.Enabled = false;
+                return;
+            }
+
+            // check Automated Updates Availability
+            foreach (IXenConnection connection in connectionList)
+            {
+                var poolOfOne = Helpers.GetPoolOfOne(connection);
+                var pool = Helpers.GetPool(connection);
+
+                if ((poolOfOne != null && poolOfOne.IsAutoUpdateRestartsForbidden) ||
+                    (pool != null && !pool.IsPoolFullyUpgraded) ||
+                    (Updates.GetUpgradeSequence(connection) == null) ||

```

Comment:
Here also, the Updates.GetUpgradeSequence(connection) is called three times. It's much more efficient to calculate it once: `var seq =  Updates.GetUpgradeSequence(connection)` and then use this instead. This means you'll probably have to split this big if check to smaller ifs and continues.